filelists/CIFAR-FS/cifar_fs_preprocessing.py

```diff
@@ -0,0 +1,490 @@
+                normalize
+            ])
+
+    def __getitem__(self, index):
+        img, label = self.data[index], self.labels[index]
+        # doing this so that it is consistent with all other datasets
+        # to return a PIL Image
+        img = Image.fromarray(img)
+        if self.transform is not None:
+            img = self.transform(img)
+        return img, label
+
+    def __len__(self):
+        return len(self.data)
+
+class CIFAR_FS_domain_shift(data.Dataset):
+
+    def __init__(self, superclass_train, super_class_test, phase="train"):
+        self.phase = phase
+        train = load_data(os.path.join(
+            _CIFAR_FS_DATASET_DIR,
+            f'cifar-by-superclass/superclass_{superclass_train}.pickle'))
+        if superclass_train != super_class_test:
+            test = load_data(os.path.join(
+                _CIFAR_FS_DATASET_DIR,
+                f'cifar-by-superclass/superclass_{superclass_train}.pickle'))
+            self.x_test = test["data"]
+            self.y_test = test["labels"]
+            self.x_train = train["data"]
+            self.y_train = train["labels"]
+        else:
+            labels = train["labels"]
+            data = train["data"]
+            random_class = labels[np.random.choice(labels.shape[0], 1, replace=False)]
+            self.x_test = labels[labels==random_class]
+            self.y_test = data[labels==random_class, :]
+            self.x_train = labels[labels!=random_class]
+            self.y_train = data[labels!=random_class, :]
+
+
+    def set_phase(self, phase):
+        assert(phase == 'train' or phase == 'test')
+        self.phase = phase
+
+    def __getitem__(self, index):
+        if self.phase == "train":
+            img, label = self.x_train[index], self.y_train[index]
+        elif self.phase == "test":
+            img, label = self.x_test[index], self.y_test[index]
+        else:
+            raise ValueError(f"phase can be either train or test but is {self.phase}")
+        img = Image.fromarray(img)
+        return img, label
+
+    def __len__(self):
+        if self.phase == "train":
+            return len(self.y_train)
+        elif self.phase == "test":
+            return len(self.y_test)
+        else:
+            raise ValueError(f"phase can be either train or test but is {self.phase}")
+
+class FewShotDataloader():
+    def __init__(self,
+                 dataset,
+                 nKnovel=5,  # number of novel categories.
+                 nKbase=-1,  # number of base categories.
+                 nExemplars=1,  # number of training examples per novel category.
+                 nTestNovel=15 * 5,  # number of test examples for all the novel categories.
+                 nTestBase=15 * 5,  # number of test examples for all the base categories.
+                 batch_size=1,  # number of training episodes per batch.
+                 num_workers=4,
+                 epoch_size=2000,  # number of batches per epoch.
```

Comment:
Isn't that a function of the number of training samples and "normal" (not the one in line 208) batch size?